sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -58,14 +58,19 @@ public RxGatewayStoreModel(
             QueryCompatibilityMode queryCompatibilityMode,
             UserAgentContainer userAgentContainer,
             GlobalEndpointManager globalEndpointManager,
-            HttpClient httpClient) {
+            HttpClient httpClient,
+            ApiType apiType) {
         this.clientContext = clientContext;
         this.defaultHeaders = new HashMap<>();
         this.defaultHeaders.put(HttpConstants.HttpHeaders.CACHE_CONTROL,
                 "no-cache");
         this.defaultHeaders.put(HttpConstants.HttpHeaders.VERSION,
                 HttpConstants.Versions.CURRENT_VERSION);
 
+        if (apiType != null){
+            this.defaultHeaders.put(HttpConstants.HttpHeaders.API_TYPE, apiType.toString());
```

Comment:
Why api type is need here, request already populating it as per your previous PR in RxDocumentClientImpl

Reply:
There is one path where the header is not populated. readCollectionAsync in RxClientCollectionCache, the request constructed there is not going through the normal populateHeaders() flow